encoder_test.go

```
@@ -65,3 +68,577 @@ func TestMatchLen(t *testing.T) {
+	t.Run("empty", func(t *testing.T) {
+		e := NewEncoder()
+		c1 := e.AppendCompress(nil, nil)
+		c2 := e.AppendCompress(nil, []byte{})
+		if len(c1) == 0 {
+			t.Fatal("expected non-empty frame for nil input")
+		}
+		if !bytes.Equal(c1, c2) {
+			t.Fatal("nil and empty should produce identical frames")
+		}
+		r := NewReader(bytes.NewReader(c1), nil)
+		defer r.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `r.Close` is not checked

(errcheck)